src/cloud-api-adaptor/test/provisioner/libvirt/provision_common.go

```diff
@@ -77,6 +78,11 @@ func NewLibvirtProvisioner(properties map[string]string) (pv.CloudProvisioner, e
 		return nil, err
 	}
 
+	container_runtime := pv.DEFAULT_CONTAINER_RUNTIME
+	if properties["container_runtime"] != "" {
+		container_runtime = properties["container_runtime"]
+	}
```

Comment:
I'm not necessarily against this code, but is it doing the same thing as https://github.com/confidential-containers/cloud-api-adaptor/blob/945ee79ac21592bb2f84e7d49990dbc81afc6124/src/cloud-api-adaptor/test/e2e/main_test.go#L134-L139? The question behind the question is do we need this code in all provisions that support cri-o?

Reply:
Yes, all provisions will require that handling as each one implement the `GetProperties` (https://github.com/confidential-containers/cloud-api-adaptor/blob/945ee79ac21592bb2f84e7d49990dbc81afc6124/src/cloud-api-adaptor/test/e2e/main_test.go#L127). I've a similar change for AWS that I should send soon.

There should be a way to refactor that to have a GetProperties returning common properties (e.g. CAA_IMAGE, KBS_IMAGE, CONTAINER_RUNTIME...) -- implemented only once -- and provisioner specific ones (e.g. libvirt_uri, ...)

Reply:
Sorry, I'm not sure I follow your reason. the docker provisioning code doesn't seem to have that default logic, but uses the shared code that I linked. What's the reason that libvirt and AWS can't use that?, or have I missed something?

Reply:
oh, sorry, now reading your first comment with proper attention, I understood your point. You are right, I will change that!

Reply:
Done!